Fixes https://www.nsnam.org/bugzilla/show_bug.cgi?id=2098

teto opened the pull request:
The documentation (https://www.nsnam.org/docs/dce/manual/html/dce-user-newapps.html) says
"Copy the executable file produced in a specified directory in the
variable environment DCE_PATH so that DCE can find it. (FIXME: to be
updated)" but wutils.py was overriding the user DCE_PATH and DCE_ROOT.

This patch prepends the user exported values for these 2 environment
variables.

thehajime commented:
I was planing to make those variables (i.e., DCE_ROOT, DCE_PATH) invisible because they sometimes made users confused.

I would like to know why those custom locations for dce binaries are useful: then this patch will get on the venue.

btw, if you would update wutils.py, you also need to update test.py as it also refers DCE_ROOT etc.

teto commented:
As you had advised me on the mailing list, I used to symlink my binaries in bin_dce but everytime I did a ./waf clean I had to redo the symlinks which is quite annoying since I see no way to automate this. Using DCE_PATH makes my programs loaded all the time and if I want to change the version of the program loaded, it's immediate to do `$ DCE_PATH="new/path" ./waf --run ` and it keeps the default configuration.
If you are ok with this, I can also update test.py (or should test.py import wsutil.py ?)

teto commented:
My guess is that environments variables are often upper case and as the variable is presented in the documentation, people assumed it was an envionrment variable they could use as they use PATH. Problem is as it gets overriden, it didn't work hence the confusion. This patch would also clear the confusion I think.

teto commented:
I switched to the "master" branch for some testing and DCE could not find my binaries anymore. I updated the patch to modify test.py as well. Hope this can get merged.

thehajime commented:
thanks for the request.

1) could you also propose a document update, i.e., to doc/source/dce-user-newapps.rst ?
2) as ns-3-dev does, updating RELEASE_NOTES would be great. this should be in 'Bugs fixed' section in the top.

teto commented:
I don't know the .rst format so I kept the changes to minimal.

thehajime commented:
Thanks. one more (two?) thing.

1) please rebase your tree to the master branch and 2) move the line of ```- Bug 2098..``` to "Release dce-dev"  in the RELEASE_NOTES. 

The bugfix will be appeared in the next release.

teto commented:
done

thehajime commented:
thanks, applied.
